### DISCUSSION TITLE
onDragEnd of a Draggable modal, the modal is reset to initial coordinates

### DISCUSSION THREAD

**Discussion by cfthody** [clauderic/dnd-kit#974] [Q&A]
I'm trying to use this library with ant design modal but I could not grasp why when dragging the modal and onDragEnd the popup is always reset to the initial coordinates. Shouldn't it be the position I have dragged as the transform property from the useDraggable return the delta coordinates?
Another question I have is that if a Wrapp the modal in a component that returns the `<DnDContext><Modal></DnDContext` it doesn't work, but if the DnDContext is in a parent component that contains the wrapped modal it works.

You can check the sandbox here: [Draggable Modal](https://codesandbox.io/s/stoic-knuth-hwrjt1?file=/src/Popup.tsx:482-492)

**Comment by mrm2234** [comment]
Hey - were you able to figure out why it wasn't dragging to the right place - I'm having a similar issue : (

**Reply by ankuratudemy** [reply]
When sending items from parent component to child component and maintaining state , you may need to set items in state everytime parent component sends it back to child, after receiving it from child component on "onDragEnd" called.:

Try this:


import { useEffect, useState } from "react";
import {
  DndContext,
  closestCenter,
  KeyboardSensor,
  PointerSensor,
  useSensor,
  useSensors,
} from "@dnd-kit/core";
import { List, ListItem, ListItemText, useTheme } from "@mui/material";
import {
  arrayMove,
  SortableContext,
  useSortable,
  sortableKeyboardCoordinates,
  sortablePointerCoordinates,
} from "@dnd-kit/sortable";


const VerticalSortableComponent = ({ options, onSort }) => {
  const [items, setItems] = useState(options);
  useEffect(() => {
    setItems(options);
  }, [options]);
  const sensors = useSensors(
    useSensor(PointerSensor),
    useSensor(KeyboardSensor, {
      coordinateGetter: sortableKeyboardCoordinates,
    })
  );

  const handleDragEnd = (event) => {
    console.log(event)
    const { active, over } = event;
    if (active.id !== over?.id) {
      const oldIndex = items.indexOf(active.id);
      const newIndex = items.indexOf(over?.id);
      console.log(oldIndex,newIndex)
      const updatedItems = arrayMove(items, oldIndex, newIndex);
      setItems(updatedItems);
      console.log(updatedItems)
      onSort(updatedItems);
    }
  };

  return (
    <DndContext
      sensors={sensors}
      collisionDetection={closestCenter}
      onDragEnd={handleDragEnd}
    >
      <SortableContext items={items} strategy={verticalListSortingStrategy}>
        <List>
          {items.map((id) => (
            <SortableItem key={id} id={id} text={id} />
          ))}
        </List>
      </SortableContext>
    </DndContext>
  );
};

const SortableItem = ({ id, text, index }) => {
  const theme = useTheme();
  const { attributes, listeners, setNodeRef, transform, transition } =
    useSortable({ id });

  return (
    <ListItem
      ref={setNodeRef}
      style={{
        transform: transform
          ? `translate3d(${transform.x}px, ${transform.y}px, 0)`
          : undefined,
        transition,
        color: theme.palette.primary.main,
        backgroundColor: theme.palette.background.main
      }}
      {...attributes}
      {...listeners}
    >
      <ListItemText primary={text} />
    </ListItem>
  );
};

// Vertical sorting strategy
const verticalListSortingStrategy = ({ index }) => {
  return {
    x: 0,
    y: index * 60, // Adjust the height of each item as needed
  };
};

export default VerticalSortableComponent;



//Above code worked for me.
